src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -146,17 +146,46 @@ C_FUNC(\Name\()_End):
 
         __PWTB_StackAlloc = __PWTB_TransitionBlock
 
+	// Make the stack operation compatible with GDB and LIBUNWIND
+	push {r7, lr}
+	mov r7, sp
+	sub sp, #((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+
         .ifnc \pushArgRegs, DoNotPushArgRegs
-                PUSH_ARGUMENT_REGISTERS
+		sub r7, #12
+		str r0, [r7]
+		str r1, [r7 , #4]
+		str r2, [r7 , #8]
+		str r3, [r7 , #12]
         .endif
 
-        PUSH_CALLEE_SAVED_REGISTERS
-        
-        alloc_stack     __PWTB_StackAlloc
+	sub r7, #36
+	str r4, [r7]
```

Comment:
As long as we save r4-r11 into the stack, that won't matter.
But, yes, using r12 instead of r7 makes the code look more clean.
Anyway, it appears that it's the custom or protocol of ARM gnueabi compilers (for making it compatible with debuggers or unwinders) to do `mov r7 sp` after `push` before `sub sp #N` (creating space for local variables). Right now, I'm blindly following what my compiler (clang 3.6 arm-linux-gnueabi) spits out with conventional C/CPP codes.